static/app/views/settings/organizationDeveloperSettings/sentryFunctionDetails.tsx

```diff
@@ -69,13 +82,48 @@ export default function SentryFunctionDetails(props: Props) {
           apiMethod={method}
           apiEndpoint={endpoint}
           model={form.current}
-          onPreSubmit={() => addLoadingMessage(t('Saving changes..'))}
+          onPreSubmit={() => {
+            updateCode();
+            addLoadingMessage(t('Saving changes..'));
+          }}
           onSubmitError={handleSubmitError}
           onSubmitSuccess={handleSubmitSuccess}
         >
           <JsonForm forms={[{title: t('Sentry Function Details'), fields: formFields}]} />
+          <Panel>
+            <PanelHeader>Write your Code Below</PanelHeader>
+            <PanelBody>
+              <Editor
+                height="40vh"
+                theme="light"
+                defaultLanguage="javascript"
+                defaultValue={defaultCode}
+                onMount={handleEditorDidMount}
```

Comment:
This seems like an anti pattern here - usually if you need to use a ref, you would just pass it in as "ref={editorRef}" instead of setting the ref in a callback function. I think they have an `onChange` callback that you can use?